### DISCUSSION TITLE
Need a quick answer to 'polarity' option

### DISCUSSION THREAD

**Discussion by fatloot** [mzmine/mzmine#1056] [General]
So I'm fairly new to MZmine and mass detection, etc. and bit of a dumb clarity question from me but I'm wondering about the 'polarity' option under the 'Set filters' button when doing Mass Metection or LC-MS: ADAP chromatagram builder.

So I understand that when doing LC-MS work you must add or subtract a proton (+H) in order to detect metabolites and your raw data will come out like this: m/z+1 [H] and m/z-1[H] aka your results will come out as 'positive' and 'negative' polarity. In MZmine3 you can choose which polarity option you want (depending on your raw data files you decide to drop in) [Any, -, +].

My question is, does this 'polarity' option in MZmine3 mean it automatically change your Mass/Charge (m/z) when you export your data (i.e. .csv file) back to the original Mass/Charge? Or does it keep the m/z charge that you originally have in the raw data from the machine when putting it into MZmine3?

Below is a more visual interpretation:

Pre-LC-MS machine: Metabolite m/z and Metabolite m/z
Post-LC-MS machine: Metabolite m/z+1[H] and Metabolite m/z-1[H]
**Post MZmine3 exported data: Metabolite m/z and Metabolite m/z**

OR

Pre-LC-MS machine: Metabolite m/z and Metabolite m/z
Post-LC-MS machine: Metabolite m/z+1[H] and Metabolite m/z-1[H]
**Post MZmine3 exported data: Metabolite m/z+1[H] and Metabolite m/z-1[H]**

Much thanks!

**Comment by robinschmid** [comment]
Welcome,
MZmine is doing a non target feature detection so the polarity does not matter in the mass detection etc steps. This is only for you to filter the scans in case your instrument supports polarity switching where you have both positive and negative mode data in one data file.

There is the ion identity molecular networking workflow where ions can be annotated. Search for documentations. 

Or you can use the local database search or targeted feature detection to specify which ions to include in the search. 
Generally there might be more adducts like sodium and ammonia in your MS data.

**Reply by fatloot** [reply]
Gotcha, thanks for the reply!

**Comment by bkieft-usa** [comment]
Hi, jumping into this thread. I am trying to use this polarity parameter in a batch parameters file via mzmine3 CLI to do exactly what you're describing in your answer @robinschmid:

#This is only for you to filter the scans in case your instrument supports polarity switching where you have both positive and negative mode data in one data file.

But when I set the polarity parameter to POSITIVE, it doesn't actually filter my FPS mzML file by positive scans only. Am I setting the parameter incorrectly? Please see my recent issue here: https://github.com/mzmine/mzmine/issues/2289

Thanks!
